### ISSUE TITLE
Gradient Descent

### ISSUE THREAD

**Issue by adhiiisetiawan** [adhiiisetiawan/pytorch-back-to-basic#11]
Optimize the loss function with respect to the parameters using the gradient descent algorithm. Build intuition for how gradient descent works from first principles.

**Comment by adhiiisetiawan** [comment]
done
